src/CustomHooksAndOtherUtils/ThemeContext.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useContext, useState } from "react"
+
+const ThemeContext = React.createContext();
+const ThemeUpdateContext = React.createContext();
+
+export function useTheme() {
+  return useContext(ThemeContext);
+}
+export function useThemeUpdate() {
+  return useContext(ThemeUpdateContext);
+}
+
+
+
+export function ThemeProvider({ children }) {
+  const [darkTheme, setDarkTheme] = useState(true);
+
+  function toggleTheme() {
+    setDarkTheme(theme => !theme);
+  }
+
+  return (
+    <ThemeContext.Provider value={darkTheme}>
```

Comment:
do you need two contexts ??? just to update the theme? make it one this is overkill